tests/test_integration_segmentation_3d.py

```diff
@@ -282,6 +282,28 @@ def test_training(self):
         np.testing.assert_allclose(repeated[0], repeated[1])
         np.testing.assert_allclose(repeated[0], repeated[2])
 
+    @TimedCall(seconds=180, daemon=False)
+    def test_timing(self):
```

Comment:
I think most of the logic in this function is duplicated to test_training, could you please combine them or extract a common internal function?